Tests/MongoSwiftTests/AsyncTestUtils.swift

```diff
@@ -39,26 +39,116 @@ extension MongoDatabase {
+extension MongoCollection {
+    fileprivate func syncDropOrFail() {
+        do {
+            try self.drop().wait()
+        } catch {
+            XCTFail("Error dropping test collection: \(error)")
         }
     }
+}
 
+extension MongoSwiftTestCase {
     internal func withTestClient<T>(options: ClientOptions? = nil, f: (MongoClient) throws -> T) throws -> T {
         let elg = MultiThreadedEventLoopGroup(numberOfThreads: 1)
         defer { elg.syncShutdownOrFail() }
         let client = try MongoClient.makeTestClient(eventLoopGroup: elg, options: options)
         defer { client.syncCloseOrFail() }
         return try f(client)
     }
+
+    /// Creates the given namespace using the given client and passes handles to it and its parent database to the given
+    /// function. After the function executes, the collection associated with the namespace is dropped.
+    ///
+    /// Note: If a collection is not specified as part of the input namespace, this function will throw an error.
+    internal func withTestNamespace<T>(
+        client: MongoClient,
+        ns: MongoNamespace? = nil,
+        options: CreateCollectionOptions? = nil,
+        _ f: (MongoDatabase, MongoCollection<Document>) throws -> T
+    ) throws -> T {
+        let ns = ns ?? self.getNamespace()
+
+        guard let collName = ns.collection else {
+            throw TestError(message: "missing collection")
+        }
+
+        let database = client.db(ns.db)
+        let collection: MongoCollection<Document>
+        do {
+            collection = try database.createCollection(collName, options: options).wait()
+        } catch let error as CommandError where error.code == 48 {
+            try database.collection(collName).drop().wait()
+            collection = try database.createCollection(collName, options: options).wait()
+        }
+        defer { collection.syncDropOrFail() }
+        return try f(database, collection)
+    }
+
+    /// Creates the given namespace and passes handles to it and its parents to the given function. After the function
+    /// executes, the collection associated with the namespace is dropped.
+    ///
+    /// Note: If a collection is not specified as part of the input namespace, this function will throw an error.
+    internal func withTestNamespace<T>(
+        ns: MongoNamespace? = nil,
+        collectionOptions: CreateCollectionOptions? = nil,
+        _ f: (MongoClient, MongoDatabase, MongoCollection<Document>) throws -> T
+    ) throws -> T {
+        return try self.withTestClient { client in
+            try self.withTestNamespace(client: client, ns: ns, options: collectionOptions) { db, coll in
+                try f(client, db, coll)
+            }
+        }
+    }
+
+    /// Captures any command monitoring events filtered by type and name that are emitted during the execution of the
+    /// provided closure. Only events emitted by the provided client will be captured.
+    internal func captureCommandEvents(
```

Comment:
It was being used elsewhere to test the "singleBatch" stuff, but now I don't think it is. I guess we might as well leave it in since we'll probably use it in the future though.